### DISCUSSION TITLE
Placing text on the top of boxes

### DISCUSSION THREAD

**Discussion by psads-git** [cetz-package/cetz#522] [Q&A] [answered]
I am using the following code to get the percentages on the top of each rectangle. Unfortunately, the percentages overlap the top line of the rectangles. Since I am a CeTZ beginner, I cannot find a solution for that problem. Could you please help me? Thanks!

```
#import "@preview/cetz:0.2.1"

#align(center,
  cetz.canvas({
    import cetz.draw: *
  
    content((0, 0), (3, 0.8), 
      box(
        align(center+horizon,
          par(justify: false)[Dataset]),
          stroke: 1pt, 
          width: 100%, 
          height: 100%, 
          inset: 1em),
      name: "dataset"
    )

    content("dataset.north",  [$100%$])

    rect((0, 2), (3, 0.8), name: "rect1")
    content("rect1.north",  [$50%$])

  })
)
```

![image](https://github.com/johannes-wolf/cetz/assets/75945439/1683385a-8696-4ffb-81ac-d33a6ffe78e3)

**Comment by psads-git** [comment]
I guess I have found a solution:

```
#import "@preview/cetz:0.2.1"

#align(center,
  cetz.canvas({
    import cetz.draw: *
  
    content((0, 0), (3, 0.8), 
      box(
        align(center+horizon,
          par(justify: false)[Dataset]),
          stroke: 1pt, 
          width: 100%, 
          height: 100%, 
          inset: 1em),
      name: "dataset"
    )

    content((v => cetz.vector.add(v, (0, 0.3)), "dataset.north"),  [$100%$])
  })
)

```

![image](https://github.com/johannes-wolf/cetz/assets/75945439/8d0c8977-482b-4702-8a94-e840ee05aee5)

**Reply by johannes-wolf** [reply]
Some tips:
- You can use `anchor: "south"` in your `content` call, to specify the origin. Together with `padding: (bottom: .3)` this should do the trick.
- You can add coordinates using the `(rel: (...), to: (...))` syntax instead of using functions with `vector.add`, the first works correctly with absolute lengths etc.
- To frame `content` you can use `frame: "rect"` together with `padding: ...`. The frame is drawn with the current stroke style

**Reply by johannes-wolf** [reply, marked as answer]
```typst
#import "@preview/cetz:0.2.1"

#align(center,
  cetz.canvas({
    import cetz.draw: *
  
    content((0, 0), (3, 0.8), 
      align(center + horizon)[Dataset],
      frame: "rect",
      name: "dataset"
    )

    content("dataset.north",  [$100%$], anchor: "south", padding: (bottom: .3))
  })
)
```
![image](https://github.com/johannes-wolf/cetz/assets/519002/951e7a00-900f-4868-9e92-afc6826b6821)

**Reply by psads-git** [reply]
Nice solution, @johannes-wolf! Thanks!
